src/main/scala/firrtl/passes/memlib/MemConf.scala

```diff
@@ -13,7 +13,22 @@ case object MaskedReadWritePort extends MemPort("mrw")
 
 object MemPort {
 
+  // All ports, make sure that any chances to all is reflected in orderedPorts below
   val all = Set(ReadPort, WritePort, MaskedWritePort, ReadWritePort, MaskedReadWritePort)
+  // This is the order that ports will render in MemConf.portsStr
+  val orderedPorts = collection.immutable
+    .Seq(
+      MaskedReadWritePort,
+      MaskedWritePort,
+      ReadWritePort,
+      WritePort,
+      ReadPort
+    )
+    .zipWithIndex
+    .toMap
+
+  // uses orderedPorts when sorting MemPorts
+  implicit def ordering: Ordering[MemPort] = Ordering.by(e => orderedPorts(e))
```

Comment:
```suggestion
  // This is the order that ports will render in MemConf.portsStr
  val ordered: Seq[MemPort] = Seq(
      MaskedReadWritePort,
      MaskedWritePort,
      ReadWritePort,
      WritePort,
      ReadPort
    )
  
   val all: Set[MemPort] = ordered.toSet

  // uses orderedPorts when sorting MemPorts
  implicit def ordering: Ordering[MemPort] = {
    val orderedPorts = ordered.zipWithIndex.toMap
    Ordering.by(e => orderedPorts(e))
  }
```

The goals of my suggestion are:
1) Keep it dry, might as well derive `all` from the ordered list
2) If we're adding a public API, I think just having the `Seq` (which is ordered) makes more sense than `Map[_, Int]`